src/univariate/continuous/weibull.jl

```diff
@@ -139,3 +139,41 @@ end
 #### Sampling
 
 rand(rng::AbstractRNG, d::Weibull) = xv(d, randexp(rng))
+
+#### Fit model
+
+"""
+    fit_mle(::Type{<:Weibull}, x::AbstractArray{T}; 
+    α0::Float64 = 1.0, maxiter::Int = 1000, tol::Float64 = 1e-16)
+
+Maximum Likelihood Estimate of `Weibull` Distribution via Newton's Method
+
+"""
+function fit_mle(::Type{<:Weibull}, x::AbstractArray{T};
+    α0::Float64 = 1.0, maxiter::Int = 1000, tol::Float64 = 1e-16) where T <: Real
+
+    ϵ = Inf
+    N = 0
+
+    α = α0
+
+    lnx = log.(x)
+    n = length(x)
+
+    while ϵ > tol && N < maxiter
+
+        xpow = x.^α
+
+        fx = sum(xpow .* lnx) / (sum(xpow)) - (1 / n) * sum(lnx) - 1 / α
+        ∂fx = (-sum(xpow .* lnx)^2 + sum(xpow) * sum((lnx.^2) .* xpow)) / (sum(xpow)^2) + 1 / α^2
+
+        αold = α
+        α -= fx / ∂fx
+
+        ϵ = abs(α - αold)
```

Comment:
These assignments might create type instabilities since their type might chnage from `Float64` to some other type. A solution would be to not initialize them with the default values `Inf` etc. before the first iteration but just set them to the computed values in the first iteration which would have to be moved outside of the loop.

Reply:
For this I removed `ϵ` and replaced it with the boolean flag `convergence` which is updated inside the loop. For the type instability in `α` I gave it a type 
```
param_type = typeof(AbstractFloat(alpha0))
α::param_type = alpha0
```
so the types of the parameters in the returned `Weibull` distribution are decided by the type passed to `alpha0`. This removes the type instabilities in the loop. If this isn't satisfactory however I can go ahead with your suggestion of writing the first iteration of loop outside the loop body.

Reply:
Unfortunately, this is not sufficient and might even fail in the case of theta: the type of alpha and theta depends also on the type of `x`. E.g., if `x` are dual numbers the parameters should be dual numbers as well even though `alpha0` is only of type `Int` or `Float64`. I think the only reliable way to fix potential type instabilities is to move the initial step outside of the loop and use it to initialize alpha (and don't enforce the type of theta).

Reply:
Hopefully the latest commit fixes this, thanks for the help and comments, here's to my first of hopefully many open source contributions :).